src/turing_utilities.jl

```diff
@@ -0,0 +1,31 @@
+using Turing
+"""
+    compute_loo(chain::Chain, data, ll_fun)
+
+Computes leave one out (loo) cross validation based on posterior distribution of model parameters.
+
+- `chain`: an MCMCChains object from Turing
+- `model`: a Turing model with data in the form of model(data)
+"""
+function compute_loo(chain::Chains, model)
+    pointwise_lls = pointwise_loglikes(chain, model)
+    psis_output = psis(pointwise_lls)
+    return compute_loo(psis_output, pointwise_lls)
+end
+
+"""
+    pointwise_loglikes(chain::Chains)
+
+Computes the pointwise log likelihoods where [d,s,c] corresponds to log likelihood of 
+evaluated at datapoint d, sample s, for chain c. 
+
+- `chain`: a chain object from MCMCChains from a Turing model
+- `model`: a Turing model with data in the form of model(data)
+"""
+function pointwise_loglikes(chain::Chains, model)
+    chain_params = MCMCChains.get_sections(chain, :parameters)
+    pwLL_dict = pointwise_loglikelihoods(model, chain_params)
+    ind_from_string(x) = parse(Int, x[3:end-1])
```

Comment:
Not sure what this is doing; can you add comments or rewrite it to be more clear?

Reply:
I left some comments. This is code that Seth provided. Unfortunately, Turing outputs Dictionaries. Another important limitation is that it only works when the log likelihoods are computed in a for loop in the model. If broadcasting is used, the log likelihoods are summed to a single value for the entire dataset.

Reply:
More confused about the line ```ind_from_string(x) = parse(Int, x[3:end-1])``` -- why those indices?

Reply:
Good question. This code that I got from Seth converted a string "y[i]" to an Int i. After looking at it closely, I realized it would not handle cases in which variable names were longer than one character. I updated the code so that it extracts the number with `split` for any variable name.